fs/dafsserver/dafsserver.cpp

```diff
@@ -1064,6 +1068,7 @@ class CRemoteRequest : public CSimpleInterfaceOf<IInterface>
             processRead(requestTree, responseMb);
         else if (activity->queryIsWriteActivity())
             processWrite(requestTree, restMb, responseMb);
+        activity->flushStatistics(stats);
```

Comment:
Who updates the bytesRead and/or bytesWritten for this ?

Reply:
depends on the activity implementation. For read bytesRead is updated via the IFileSerialStreamCallback callback,
For write, bytesWritten it's simpler, updated as write is called.

Reply:
ok.
```
class CRemoteWriteBaseActivity : 
    virtual void write(size32_t sz, const void *rowData) override
    {
        throwUnexpected(); // method should be implemented in derived classes.
    }
    virtual void flushStatistics(CClientStats &stats) override
    {
        // NB: will be called by same thread that is writing.
        stats.addWrite(bytesWritten);
        bytesWritten = 0;
    }
```
And
```
class CRemoteDiskWriteActivity :
    virtual void write(size32_t sz, const void *rowData) override
    {
        checkOpen();
        iFileIOStream->write(sz, rowData);
        bytesWritten += sz;
    }
```

Reply:
```
class CRemoteDiskBaseActivity : 
    virtual void flushStatistics(CClientStats &stats) override
    {
        throwUnexpected();
    }
```

Reply:
ok.  I guess I was originally looking for a base flushStatistics() that updated both reads and writes.